src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/CommunityAPIOffenderService.kt

```diff
@@ -128,36 +126,30 @@ class CommunityAPIOffenderService(
       .bodyToMono(OffenderIdentifiersResponse::class.java)
   }
 
-  fun getResponsibleOfficer(crn: String): ResponsibleOfficer {
-    val offenderManagersPath = UriComponentsBuilder.fromPath(offenderManagersLocation)
+  fun getResponsibleOfficerDetails(crn: String): OfficerDetails? {
+    val officerDetailsPath = UriComponentsBuilder.fromPath(responsibleOfficerLocation)
       .buildAndExpand(crn)
       .toString()
 
-    val responsibleOfficers = communityApiClient.get(offenderManagersPath)
+    val responsibleOfficer = communityApiClient.get(officerDetailsPath)
       .retrieve()
-      .bodyToFlux(OffenderManagerResponse::class.java)
-      .filter { it.isResponsibleOfficer == true }
-      .map { ResponsibleOfficer(it.staff?.forenames?.substringBefore(' '), it.staff?.email, it.staffId, it.staff?.surname) }
-      .collectList()
+      .bodyToMono(OfficerDetails::class.java)
+      .onErrorResume(WebClientResponseException::class.java) { e ->
+        when (e.statusCode) {
+          // not all delius users are staff
+          HttpStatus.NOT_FOUND -> Mono.empty()
+          else -> Mono.error(e)
+        }
+      }
       .block()
 
-    // as with many community API endpoints, we have a few assumptions about the data.
-    // if there are no ROs, we can't recover. if there are more than one, we just take
-    // the first.
-    responsibleOfficers.size.let {
-      when {
-        it == 0 -> telemetryService.reportInvalidAssumption(
-          "service users always have a responsible officer",
-          mapOf("crn" to crn),
-          recoverable = false,
-        )
-        it > 1 -> telemetryService.reportInvalidAssumption(
-          "service users only have one responsible officer",
-          mapOf("crn" to crn, "numberOfResponsibleOfficers" to it.toString()),
-        )
-      }
+    if (responsibleOfficer == null) {
```

Comment:
I think we could use the responsibleOfficer flag here `|| !responsibleOfficer.communityOfficer.responsibleOfficer` to check if they are the responsible officer

Reply:
@marcocolluramoj I think we need to do both. Something like this 

`    if (officerDetails == null || !(officerDetails.communityOfficer.responsibleOfficer)) {
      telemetryService.reportInvalidAssumption(
        "service users always have a responsible officer",
        mapOf("crn" to crn),
        recoverable = false,
      )
    }`

The reason is we need to throw Invalid Assumption error when there is 404 as well

Reply:
Sure, I agree